### PR TITLE
Roll out uniform PR template

### LINKED ISSUE #558: Issue/PR templates for all repos
### What?

> Additional work should be done to roll out consistent issue/PR templates for all repos.

_Originally posted by @kenodegard in https://github.com/conda/infra/issues/515#issuecomment-1145228713_

### Description

We want issue forms for bugs, features/enhancements, etc.

Since we can only have one PR template it needs to be a checklist of actions the contributor needs to satisfy.

### Tasks
- [x] bug issue form #567
- [x] feature issue form #567
- [X] #530
- [ ] #580

### PR DESCRIPTION
**Pull request by kenodegard**
Last outstanding task for https://github.com/conda/infra/issues/558.

These PR templates are derived from the PR template already [defined for conda](https://github.com/conda/conda/blob/master/.github/PULL_REQUEST_TEMPLATE.md).

Note that we define two different PR templates:

- **Projects with releases.** Includes the checklist for news snippets, updated tests, and updated docs. While news snippets are currently only used for conda/conda-build releases it's not unreasonable for us to use them for all conda org releases. Hopefully, this change will allow us to continue standardizing how releases are done across the entire conda org.
- **Projects without releases.** No checklist defined. Currently only pertains to [conda/infra](https://github.com/conda/infra) and [conda/issue-tracker](https://github.com/conda/issue-tracker).

Resolves #558

### CODE REVIEW

File: .github/workflows/sync.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ on:
       - .github/ISSUE_TEMPLATE/0_bug.yml
       - .github/ISSUE_TEMPLATE/1_feature.yml
       - .github/ISSUE_TEMPLATE/epic.yml
+      - .github/PULL_REQUEST_TEMPLATE.md
```
Comment (kenodegard) [inline .github/workflows/sync.yml:20]:
```suggestion
      - .github/PULL_REQUEST_TEMPLATE_w_releases.md
      - .github/PULL_REQUEST_TEMPLATE_wo_releases.md
```

**Review by beeankha — APPROVED** [review]
👍

File: .github/sync.yml [hunk 2]
```diff
@@ -42,3 +42,31 @@ group:
...
+      # PR template
+      - source: .github/PULL_REQUEST_TEMPLATE_w_releases.md
+        dest: .github/PULL_REQUEST_TEMPLATE_w_releases.md
+  # projects without releases
+  - repos: |
+      conda/issue-tracker
+    files:
+      # PR template
+      - source: .github/PULL_REQUEST_TEMPLATE_wo_releases.md
+        dest: .github/PULL_REQUEST_TEMPLATE_wo_releases.md
```
Comment (kenodegard) [inline .github/sync.yml:72]:
```suggestion
      # PR template
      - source: .github/PULL_REQUEST_TEMPLATE_w_releases.md
        dest: .github/PULL_REQUEST_TEMPLATE.md
  # projects without releases
  - repos: |
      conda/issue-tracker
    files:
      # PR template
      - source: .github/PULL_REQUEST_TEMPLATE_wo_releases.md
        dest: .github/PULL_REQUEST_TEMPLATE.md
```

**Review by jezdez — APPROVED** [review]
This looks good in general, thank you for doing this!

Could I interest you in using a directory for all current (and future) pull request templates, e.g. `.github/PULL_REQUEST_TEMPLATE/with_releases.md` to match what we have with the `.github/ISSUE_TEMPLATE/*.yml` files?

**Pull request merged**